src/core/core/global_config.cpp

```diff
@@ -341,7 +341,19 @@ network_server_config::network_server_config(int p, rpc_channel c)
 
 bool network_server_config::operator < (const network_server_config& r) const
 {
-    return port < r.port || (port == r.port && channel < r.channel);
+    if (port != r.port)
+    {
+        return port < r.port;
+    }
+    if (channel != r.channel)
+    {
+        return channel < r.channel;
+    }
+    if (hdr_format != r.hdr_format)
+    {
```

Comment:
each network can only support one single header format, therefore header format cannot be used as part of the sorting key